nntrainer/layers/common_properties.cpp

```diff
@@ -48,6 +48,33 @@ void FilePath::set(const std::string &v) {
 
 std::ifstream::pos_type FilePath::file_size() { return cached_pos_size; }
 
+bool TensorShape::isValid(const std::string &v) const {
+  // Tensor dimension does not allow the 0 & number of dimension shuld be less
+  // than MAXDIM 4. Also the white space will be ignored.
+
+  static std::regex allowed("[1-9:]*");
+  auto check_num_tensors = [](const std::string &value) {
+    auto values = split(value, std::regex("\\s*\\:\\s*"));
+    if (values.size() > TensorDim::MAXDIM)
```

Comment:
also, check that each split value inside the values array is a non-empty string. For example,  "1::1"?

Reply:
yes. I missed it. Thanks.